s3torchconnector/tst/e2e/dcp/test_e2e_s3_file_system.py

```diff
@@ -4,19 +4,315 @@
 import pytest
 import torch
 import torch.distributed.checkpoint as dcp
+
+import torch.distributed as dist
 from torch.distributed.checkpoint import CheckpointException
+from torch.distributed.checkpoint._fsspec_filesystem import (
+    FsspecWriter,
+    FsspecReader,
+)
+import torch.multiprocessing as mp
 
 from s3torchconnector.dcp import S3StorageWriter, S3StorageReader
+from s3torchconnector._s3client import S3Client
+from s3torchconnector._s3dataset_common import parse_s3_uri
+import os
+
+
+def setup(rank, world_size):
+    os.environ["MASTER_ADDR"] = "localhost"
+    os.environ["MASTER_PORT"] = "12355"
+    dist.init_process_group("gloo", rank=rank, world_size=world_size)
+
+
+def cleanup():
+    dist.destroy_process_group()
+
+
+def run(
+    rank,
+    world_size,
+    threads,
+    region,
+    s3_path_s3storagewriter,
+    s3_path_fsspec,
+    test_data,
+):
+    print(f"Running on rank {rank}.")
+
+    setup(rank, world_size)
+    # Save using S3StorageWriter
+    dcp_save(
+        test_data,
+        S3StorageWriter(
+            region=region,
+            thread_count=threads,
+            path=s3_path_s3storagewriter,
+            overwrite=True,
+        ),
+    )
+    # Save using FsspecWriter
+    dcp_save(
+        test_data,
+        FsspecWriter(
+            path=s3_path_fsspec,
+            overwrite=True,
+            sync_files=False,
+        ),
+    )
+    cleanup()
+
+
+def multi_process_dcp_save_load(world_size, thread_count, checkpoint_directory):
+    region = checkpoint_directory.region
+    s3_path_s3storagewriter = f"{checkpoint_directory.s3_uri}checkpoint_s3storagewriter"
+    s3_path_fsspec = f"{checkpoint_directory.s3_uri}checkpoint_fsspec"
+    test_data = {
+        "tensor1": torch.randn(10, 10),
```

Comment:
could we parametrize `tensor1` here to have different values for test? see https://github.com/awslabs/s3-connector-for-pytorch/blob/main/s3torchconnector/tst/e2e/test_e2e_s3checkpoint.py#L11-L16 as example
or at least make it bigger?

Reply:
okay